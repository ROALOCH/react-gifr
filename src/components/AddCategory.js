import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim().length > 0) {
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit} className='form-group'>
			<label htmlFor='gifSearch' className='mb-2'>
				Â¿QuÃ© deseas buscar?
				<span> </span>
				<span role='img' aria-label='thinking face emoji'>
					ðŸ¤”
				</span>
			</label>
			<input
				id='gifSearch'
				type='text'
				value={inputValue}
				onChange={handleInputChange}
				className='form-control'
			/>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
