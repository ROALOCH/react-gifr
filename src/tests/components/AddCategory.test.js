import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCategory/>', () => {
	const setCategories = jest.fn();
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

	test('Debe mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de cambiar la caja de texto', () => {
		const input = wrapper.find('#gifSearch');
		const value = 'Hola Mundo';

		input.simulate('change', { target: { value: value } });
		expect(wrapper.find('p').text().trim()).toBe(value);
	});

	test('NO debe de postear la informaciÃ³n en el submit', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).not.toHaveBeenCalled();
	});

	test('Debe de llamar setCategories() y limpiar la caja de texto', () => {
		const value = 'Hola Mundo';

		wrapper.find('input').simulate('change', { target: { value: value } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategories).toHaveBeenCalledTimes(1);
		expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
