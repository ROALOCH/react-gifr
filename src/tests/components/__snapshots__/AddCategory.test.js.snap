// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe mostrar el componente correctamente 1`] = `
<form
  className="form-group"
  onSubmit={[Function]}
>
  <p />
  <label
    for="gifSearch"
  >
    Ingresa tu b√∫squeda
  </label>
  <input
    className="form-control"
    id="gifSearch"
    onChange={[Function]}
    placeholder="Ingresa lo que quieres buscar"
    type="text"
    value=""
  />
</form>
`;
